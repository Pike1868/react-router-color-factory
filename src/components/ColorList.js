import React from "react";
import { Link } from "react-router-dom";
import NavBar from "./NavBar";

/**
 * ColorsList Component:
 *
 * Renders NavBar and a list of colors, each represented as a link. Clicking on a link navigates user to a dedicated page for that color.
 *
 * Props:
 * colorsArr (array): Array of color objects
 *
 */

const ColorList = ({ colorsArr }) => {
  return (
    <div style={{ textAlign: "center" }}>
      <NavBar />
      <p style={{ margin: "30px" }}>Please select a color:</p>
      <ul
        style={{
          display: "flex",
          flexDirection: "column",
          justifyContent: "center",
        }}
      >
        {colorsArr.map((color) => (
          <Link key={color.name} to={`/colors/${color.name}`}>
            {color.name}
          </Link>
        ))}
      </ul>
    </div>
  );
};

export default ColorList;
