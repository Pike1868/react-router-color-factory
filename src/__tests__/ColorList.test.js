import React from "react";
import { render, screen } from "@testing-library/react";
import ColorList from "../components/ColorList";
import { MemoryRouter } from "react-router-dom";

const initialColors = [
  { name: "red", hex: "#ff0000" },
  { name: "green", hex: "#008000" },
  { name: "blue", hex: "#0000ff" },
];

//Smoke Test
test("renders ColorList without crashing", () => {
  render(
    <MemoryRouter>
      <ColorList colorsArr={initialColors} />
    </MemoryRouter>
  );
});

//Snapshot Test
test("ColorList component matches snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <ColorList colorsArr={initialColors} />
    </MemoryRouter>
  );

  expect(asFragment()).toMatchSnapshot();
});

test("displays ColorList component msg", () => {
  render(
    <MemoryRouter>
      <ColorList colorsArr={initialColors} />
    </MemoryRouter>
  );
  const msg = screen.getByText("Please select a color:");
  expect(msg).toBeInTheDocument();
});
