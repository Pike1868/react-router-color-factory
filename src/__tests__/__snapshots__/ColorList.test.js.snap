// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorList component matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="text-align: center;"
  >
    <nav
      style="padding: 50px; background-color: darkgray; text-align: center;"
    >
      <h3>
        Welcome to Color Factory
      </h3>
      <a
        href="/colors/new"
        style="font-size: 1.5rem;"
      >
        Add a color
      </a>
    </nav>
    <p
      style="margin: 30px;"
    >
      Please select a color:
    </p>
    <ul
      style="display: flex; flex-direction: column; justify-content: center;"
    >
      <a
        href="/colors/red"
      >
        red
      </a>
      <a
        href="/colors/green"
      >
        green
      </a>
      <a
        href="/colors/blue"
      >
        blue
      </a>
    </ul>
  </div>
</DocumentFragment>
`;
